RELEASE_PROCESS.md

```diff
@@ -0,0 +1,54 @@
+# Release Process
+
+## Goals
+
+* Do not explicitly expose Compose specification versions to users
+* Keep the process simple and light
+
+## Process
+
+### Requesting a release
+
+To trigger the release process, the requester MUST open a pull request against
+this repository with an updated [CHANGELOG.md](./CHANGELOG.md).
+Anyone can request a release of the specification.
+
+A section MUST be added to the file with the following format:
+
+```markdown
+## date (commit digest)
+
+- list of changes
+```
+
+The date:
+* MUST be of the form `YYYY-MM-DD`
+* SHOULD be the same as when the pull request was opened
+
+The commit digest:
+* MUST be of the same form as that given by the command `git rev-parse HEAD`
+* MUST be a valid commit digest on the default branch (currently `master`)
+* MUST be at the point that the release requester would like the release to be
+  made
+
+The list of changes:
+* MUST be a bullet point list
+* MUST list all impactful changes to the specification since the last release
+
+### Approving a release
+
+To approve a release, a simple majority of maintainers MUST approve the pull
```

Comment:
Note to reviewers, I put a simple majority here. Do we want to make it less than that?

Reply:
How do we define maintainers and how many does the majority equal to?

Reply:
We currently have this list: https://github.com/compose-spec/compose-spec/blob/master/OWNERS.md

We should discuss how we'd like to define maintainership in a community meeting using our [GOVERNANCE doc](https://github.com/compose-spec/compose-spec/blob/master/GOVERNANCE.md) as a starting point.

Reply:
Sounds good - added it to the agenda of an upcoming meeting 🙂